### PR TITLE
Uploading work for lesson04

### PR DESCRIPTION
**Pull request by JonCracolici**
Please take a look and let me know if you feel my programs are ready for grading.

### CODE REVIEW

File: students/Cracolici_Jon/lesson04/JonCracolici_Kata.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    opentext = opentext.lower()
+    opentext = opentext.splitlines()
+    #Scrubbing punctuation from text
+    translationtable = dict.fromkeys(map(ord, '-,.!@#?:;%&$)(""'), ' ')
+    fstring = ''
+    for item in opentext:
+        fstring += ' ' + item.translate(translationtable)
+    #Whitespace handling
+    translation_table2 = dict.fromkeys(map(ord, '   '), ' ')
+    opentext = fstring.translate(translationtable2)
```
Comment (charlesrobison) [inline students/Cracolici_Jon/lesson04/JonCracolici_Kata.py]: Typo here - should be translation_table2

File: students/Cracolici_Jon/lesson04/JonCracolici_Kata.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+def export_book(book):
+    """This function takes a string and makes a document with it."""
+    bookname = input('What do you want to name the file?')
+    bookname+='.txt'
+    outfile = open(bookname, 'w')
+    outfile.write(book)
+    outfile.close()
+
+#The following line executes all the above for the Adventures of Sherlock Holmes
+export_book(new_book(build_dict(import_parse_text('sherlock.txt'))))
```
Comment (charlesrobison) [inline students/Cracolici_Jon/lesson04/JonCracolici_Kata.py]: Works well!

File: students/Cracolici_Jon/lesson04/JonCracolici_Kata.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+#I will need 4 basic functions: a way to import text, a way to parse it,
+#a way to build the Kata dict, and a way to use the Kata dict to create 
+#and write a new file. Lets go!
+
+import random as rand
+def import_parse_text(file):
+    """This function takes a text file, opens it and parses it into a list of lower case words."""
+    #File import portion
+    localfile = open(file, 'r')
```
Comment (charlesrobison) [inline students/Cracolici_Jon/lesson04/JonCracolici_Kata.py]: In Python you'll want all of these to be snake_case just like your function names: local_file

File: students/Cracolici_Jon/lesson04/JonCracolici_L4_Mailroom2.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    print('2) Send a thank you to a current donor for a new gift')
+    print('3) Send a thank you to a first time donor')
+    print('4) Return to the start menu')
+def saty_user_input():
+    """Collects user input. I separate this for easier error handling in future."""
+    task = input('Please type the number of your selection.')
+    return task
+def nd_control(dB):
+    """Function that calls functions to create and add a new donor to database, print letter to screen."""
+    newdonordict = donor_input(dB)
```
Comment (charlesrobison) [inline students/Cracolici_Jon/lesson04/JonCracolici_L4_Mailroom2.py]: snake_case for all of this in each function:  new_donor_dict

File: students/Cracolici_Jon/lesson04/JonCracolici_L4_Mailroom2.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    return dB#, start_control(dB)
+def ud_control(dB):
+    """Function that calls functions to update a donor in the database, print letter to screen."""
+    updonordict = donor_input(dB)
+    if updonordict == False:
+        return False
+    #will add try except to check for donor actually being new in L5
+    update_donor(updonordict, dB)
+    note_gen(dB[updonordict['donornamekey']])
+    return dB#, start_control(dB)
```
Comment (charlesrobison) [inline students/Cracolici_Jon/lesson04/JonCracolici_L4_Mailroom2.py]: Note that if the user selects an existing donor, but doesn't input the correct name the program breaks.  Suggest an exception for such cases.

File: students/Cracolici_Jon/lesson04/JonCracolici_L4_Mailroom2.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+			#return db
+#
+#
+#This is the beginning of the executed program.
+	
+if __name__=="__main__":
+
+    startmenulogic = {'1': main_2, '2': create_report, '3': mass_mail, '4': quit_program}
+    satymenulogic = {'1': show_current_names, '2': ud_control, '3': nd_control, '4': quit_saty}
+    main()
```
Comment (charlesrobison) [inline students/Cracolici_Jon/lesson04/JonCracolici_L4_Mailroom2.py]: Works quite well - nice job!

File: students/Cracolici_Jon/lesson04/JonCracolici_Kata.py (hunk 3 above)
Comment (natasha-aleksandrova) [reply students/Cracolici_Jon/lesson04/JonCracolici_Kata.py]: prefer to use `with` when working with files

**Pull request merged**